### PR TITLE
C wrap c encrypt

### PR DESCRIPTION
**Pull request by gamringer**
@vjardin you can review this.

### CODE REVIEW

File: pkcs11module.c [hunk 1]
```diff
@@ -1778,6 +1778,75 @@ PHP_METHOD(Module, C_Wrap) {
...
+    CK_ATTRIBUTE_PTR templateObj;
+    parseTemplate(&template, &templateObj, &templateItemCount);
+
+    pkcs11_object *objval = Z_PKCS11_P(ZEND_THIS);
+    pkcs11_session_object *sessionobjval = Z_PKCS11_SESSION_P(session);    
+    pkcs11_mechanism_object *mechanismobjval = Z_PKCS11_MECHANISM_P(mechanism);
+
+    if (mechanismobjval->mechanism.mechanism == 0) {
+        zend_throw_exception(zend_ce_exception, "Invalid mechanism", 0);
+        return ;
```
Comment (vjardin) [inline pkcs11module.c:1827]: missing Template free memory

**Review by vjardin — CHANGES_REQUESTED** [review]

File: pkcs11module.c [hunk 2]
```diff
@@ -1523,6 +1527,331 @@ PHP_METHOD(Module, C_Verify) {
...
+        sessionobjval->session,
+        &mechanismobjval->mechanism,
+        hKey,
+        hTargetKey,
+        hCiphertext,
+        &ciphertextLen
+    );
+    if (rv != CKR_OK) {
+        pkcs11_error(rv, "Unable to wrap");
+        return;
```
Comment (vjardin) [inline pkcs11module.c:1777]: missing free hCiphertext

**Review by vjardin — CHANGES_REQUESTED** [review]

**Comment by gamringer** [comment]
@vjardin have you checked everything you wanted to check ?

**Comment by vjardin** [comment]
Yes LGTM let's apply it.

**Pull request merged**